include/aspect/assembly.h

```diff
@@ -197,6 +202,11 @@ namespace aspect
            */
           MaterialModel::MaterialModelInputs<dim> explicit_material_model_inputs;
           MaterialModel::MaterialModelOutputs<dim> explicit_material_model_outputs;
+
+          /**
+           * Heating model outputs computed at the current linearization point.
+           */
+          HeatingModel::HeatingModelOutputs heating_model_outputs;
```

Comment:
Yes, and #775 even adds a bit more weight. But #783 suggests to remove some now unnecessary things. And we could even go further and take this class apart, into one class that is designed as scratch for the AdvectionSystem (mostly using 'current' values), and one that is used as scratch for the residual computation, which is now handled in a separate loop over cells. This new one would mostly use 'old' or 'explicit' values (named AdvectionSystemResidual?). Do you think it is worth to make this class smaller?